### PR TITLE
[BACKLOG-35624] CSRF - /j_spring_security_check - Explicit Login

### PR DESCRIPTION
**Pull request by dcleao**
Please ignore the commit [[BACKLOG-35003] Change security-web to latest version](https://github.com/pentaho/pentaho-platform/commit/31b11327f582830a6c4488b7f3f65920f9d5b476), which belongs to a base PR, https://github.com/pentaho/pentaho-platform/pull/5075, still not merged.

Related PRs:
- https://github.com/pentaho/pentaho-platform-ee/pull/1527
- https://github.com/pentaho/pentaho-commons-gwt-modules/pull/810
- https://github.com/pentaho/pentaho-kettle/pull/8300

### CODE REVIEW

File: extensions/src/main/java/org/pentaho/platform/web/http/security/CsrfValidationAuthenticationException.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * Copyright (c) 2022 Hitachi Vantara. All rights reserved.
+ */
+
+package org.pentaho.platform.web.http.security;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import org.springframework.security.access.AccessDeniedException;
```
Comment (sVarma1992) [inline extensions/src/main/java/org/pentaho/platform/web/http/security/CsrfValidationAuthenticationException.java:21]: this import might not be needed, can be removed.

File: extensions/src/test/java/org/pentaho/platform/web/http/security/CsrfProtectedUsernamePasswordAuthenticationFilterTest.java [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+    assertNull( filter.getCsrfProcessor() );
+  }
+
+  @Test
+  public void testConstructorWithGivenCsrfProcessorRespectsIt() {
+
+    CsrfProcessor csrfProcessorMock = mock( CsrfProcessor.class );
+
+    // ---
```
Comment (sVarma1992) [inline extensions/src/test/java/org/pentaho/platform/web/http/security/CsrfProtectedUsernamePasswordAuthenticationFilterTest.java:70]: Is there a reason for adding this "//---" ?

File: extensions/src/main/java/org/pentaho/platform/web/http/security/CsrfValidationAuthenticationException.java (hunk 1 above)
Comment (dcleao) [reply extensions/src/main/java/org/pentaho/platform/web/http/security/CsrfValidationAuthenticationException.java:21]: Well spotted!

File: extensions/src/test/java/org/pentaho/platform/web/http/security/CsrfProtectedUsernamePasswordAuthenticationFilterTest.java (hunk 2 above)
Comment (dcleao) [reply extensions/src/test/java/org/pentaho/platform/web/http/security/CsrfProtectedUsernamePasswordAuthenticationFilterTest.java:70]: It's a matter of organization and style.
I use that, albeit not always, I admit, to separate a test's main sections, when they're clearly defined:
```java
Preparation Code
// ---
Test Code
// ---
Validation Code
```

**Review by sVarma1992 — APPROVED** [review]
Approving the changes

**Pull request merged**